tests/common/plugins/conditional_mark/__init__.py

```diff
@@ -85,6 +86,35 @@ def load_conditions(session):
 
     return conditions_list
 
+def read_asic_name(hwsku):
+    '''
+    Get asic generation name from file 'ansible/group_vars/sonic/variables'
+
+    Args:
+        hwsku (str): Dut hwsku name
+
+    Returns:
+        str or None: Return the asic generation name or None if something went wrong or nothing found in the file.
+
+    '''
+    asic_name_file = os.path.dirname(__file__) + ASIC_NAME_PATH
+    try:
+        with open(asic_name_file) as f:
+            asic_name = yaml.safe_load(f)
+            logger.info(asic_name)
+
+        for key, value in asic_name.items():
+            if ('td' not in key) and ('th' not in key):
+                asic_name.pop(key)
+
+        for name, hw in asic_name.items():
+            if hwsku in hw:
+                return name.split('_')[1]
+
+        return None
```

Comment:
Is it better to return "unknown" such as the function `get_asic_name` does in `sonic.py`?

Reply:
It will be better.